src/apps/fuzzers/fuzzerGridDisk.c

```diff
@@ -36,8 +36,11 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
 
     int64_t sz;
     H3Error err = H3_EXPORT(maxGridDiskSize)(args->k, &sz);
-    if (err || sz > MAX_GRID_DISK_SIZE) {
-        // Can't allocate
+    if (err || sz > MAX_GRID_DISK_SIZE || sz < 0) {
```

Comment:
For my info - is any corresponding check needed in `gridDisk` itself for bad input?

Reply:
Partially yes. If the `k` value is such that an overflow will occur, we need to guard against it. There are however `k` values which are valid but simply too large to practically use, which I don't think we should guard against. We should instead document the time complexity so users can be aware of this.